pystarport/pystarport/cosmoscli.py

```diff
@@ -0,0 +1,761 @@
+    ):
+        self.data_dir = data_dir
+        if chain_id is None:
+            self._genesis = json.load(open(self.data_dir / "config" / "genesis.json"))
+            self.chain_id = self._genesis["chain_id"]
+        else:
+            self.chain_id = chain_id
+        self.node_rpc = node_rpc
+        self.raw = ChainCommand(cmd)
+        self.leger_button = LedgerButton(zemu_address, zemu_button_port)
+        self.output = None
+        self.error = None
+
+    def node_id(self):
+        "get tendermint node id"
+        output = self.raw("tendermint", "show-node-id", home=self.data_dir)
+        return output.decode().strip()
+
+    def create_account(self, name, mnemonic=None):
+        "create new keypair in node's keyring"
+        if mnemonic is None:
+            output = self.raw(
+                "keys",
+                "add",
+                name,
+                home=self.data_dir,
+                output="json",
+                keyring_backend="test",
+            )
+        else:
+            output = self.raw(
+                "keys",
+                "add",
+                name,
+                "--recover",
+                home=self.data_dir,
+                output="json",
+                keyring_backend="test",
+                stdin=mnemonic.encode() + b"\n",
+            )
+        return json.loads(output)
+
+    def create_account_ledger(self, name):
+        "create new ledger keypair"
+
+        def send_request():
+            try:
+                self.output = self.raw(
+                    "keys",
+                    "add",
+                    name,
+                    "--ledger",
+                    home=self.data_dir,
+                    output="json",
+                    keyring_backend="test",
+                )
+            except Exception as e:
+                self.error = e
+
+        t = threading.Thread(target=send_request)
+        t.start()
+        time.sleep(3)
+        for _ in range(0, 3):
+            self.leger_button.press_right()
+            time.sleep(0.2)
+        self.leger_button.press_both()
+        t.join()
+        if self.error:
+            raise self.error
+        return json.loads(self.output)
+
+    def init(self, moniker):
+        "the node's config is already added"
+        return self.raw(
+            "init",
+            moniker,
+            chain_id=self.chain_id,
+            home=self.data_dir,
+        )
+
+    def validate_genesis(self):
+        return self.raw("validate-genesis", home=self.data_dir)
+
+    def add_genesis_account(self, addr, coins, **kwargs):
+        return self.raw(
+            "add-genesis-account",
+            addr,
+            coins,
+            home=self.data_dir,
+            output="json",
+            **kwargs,
+        )
+
+    def gentx(self, name, coins, min_self_delegation=1):
+        return self.raw(
+            "gentx",
+            name,
+            coins,
+            amount=coins,
```

Comment:
this workaround is no longer needed in rc7

Reply:
@tomtau in rc7 will it use the `amount` argument or the `amount` flag?

Reply:
argument

Reply:
Thanks, updated.